pytorch_lightning/trainer/training_loop.py

```diff
@@ -400,23 +403,21 @@ def train(self):
                     self.reduce_lr_on_plateau_scheduler.step(val_loss)
 
                 if self.max_steps and self.max_steps == self.global_step:
-                    self.main_progress_bar.close()
-                    model.on_train_end()
-                    self.on_train_end()
+                    self.run_training_teardown()
                     return
 
                 # early stopping
                 met_min_epochs = epoch >= self.min_epochs - 1
                 met_min_steps = self.global_step >= self.min_steps if self.min_steps else True
 
+                # TODO wrap this logic into the callback
```

Comment:
do you want to do it in this PR?

Reply:
this callback is tricky because it's currently returning a value to the training loop which is used to determine whether or not the trainer should stop. if we now wrap the `early_stop_callback` into the list of other callbacks, this becomes infeasible - so we'd need to figure out a different way to handle this case. need to give it more thought.